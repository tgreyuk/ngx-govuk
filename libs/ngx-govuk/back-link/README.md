# ngx-govuk/back-link

Secondary entry point of `ngx-govuk`. It can be used by importing from `ngx-govuk/back-link`.
