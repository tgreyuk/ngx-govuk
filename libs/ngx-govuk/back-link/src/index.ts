export * from './back-link.directive';
