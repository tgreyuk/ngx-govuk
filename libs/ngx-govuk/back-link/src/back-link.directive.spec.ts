import { Component } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NgxGovukBackLink } from './back-link.directive';

@Component({
  standalone: true,
  imports: [NgxGovukBackLink],
  template: `<a ngxGovukBackLink>Default</a>
    <a ngxGovukBackLink [inversed]="true">Inversed</a>`,
})
class TestHostComponent {}

describe('NgxGovukBackLink', () => {
  let component: TestHostComponent;
  let fixture: ComponentFixture<TestHostComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [NgxGovukBackLink, TestHostComponent],
    });

    fixture = TestBed.createComponent(TestHostComponent);
    fixture.detectChanges();
    component = fixture.componentInstance;
  });
  it('should create an instance', () => {
    expect(component).toBeTruthy();
  });

  it('should render markup', () => {
    expect(fixture).toMatchSnapshot();
  });
});
