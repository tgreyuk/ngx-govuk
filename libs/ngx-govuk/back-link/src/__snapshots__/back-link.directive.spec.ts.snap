// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgxGovukBackLink should render markup 1`] = `
<ng-component>
  <a
    class="govuk-back-link"
    ngxgovukbacklink=""
  >
    Default
  </a><a
    class="govuk-back-link govuk-back-link--inverse"
    ngxgovukbacklink=""
  >
    Inversed
  </a>
</ng-component>
`;
